tests/expectations/metrics/column_map_metrics/test_unexpected_indices_and_query.py

```diff
@@ -305,7 +310,159 @@ def test_sa_unexpected_index_query_metric_without_id_pk(sa, animal_table_df):
         "unexpected_condition": unexpected_columns_metric,
         "table.columns": table_columns_metric,
     }
-    with pytest.raises(MetricResolutionError):
-        results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
-            metrics_to_resolve=(unexpected_index_query,), metrics=metrics
+    results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_index_query,), metrics=metrics
+    )
+    for (key, val) in results.items():
+        assert (
+            val == "SELECT animals \n"
+            "FROM test \n"
+            "WHERE animals IS NOT NULL AND (animals NOT IN ('cat', 'fish', 'dog'));"
+        )
+
+
+@pytest.mark.unit
+def test_spark_unexpected_index_list_metric_with_id_pk(
+    spark_session, animal_table_df, metric_value_kwargs_complete
```

Comment:
By nature of having a Spark session, I don't think we can call this a unit test - can we mark as integration?

Reply:
done